API/Helpers/KoreaderHelper.cs

```diff
@@ -0,0 +1,94 @@
+using API.DTOs.Koreader;
+using API.DTOs.Progress;
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace API.Helpers;
+
+public static class KoreaderHelper
+{
+    /// <summary>
+    /// Hashes the document according to a custom Koreader algorithm
+    /// </summary>
+    /// <remarks>The hashing algorithm is relatively quick as it only hashes ~10,000 bytes for the biggest of files.</remarks>
+    /// <param name="filePath">The path to the file to hash</param>
+    public static string HashContents(string filePath)
+    {
+
+        if (string.IsNullOrEmpty(filePath))
+            return null;
+
+        using (var file = File.OpenRead(filePath))
```

Comment:
This presents an interesting problem. 
KoreaderHelper is currently a static class. In order to add DirectoryService as a member of KoreaderHelper, I have to make it a concrete type. If KoreaderHelper is a concrete type, then it needs to be injected into MangaFileBuilder using dependency injection. MangaFileBuilder cannot be instantiated in the dependency tree (or composition root. Whatever you call it) since it is supposed to be instantiated ad hoc.
I can add a static member to KoreaderHelper and insert a DirectoryService object in, but that feels wrong. Do you have an alternative idea for how to achieve the mocking capability?

Reply:
Hmm, it all comes down to the unit tests. I needed to use a mocked filesystem for a lot of the unit tests to emulate last write times and whatnot, but technically, if we have some non-copywrited files in TestData, we can avoid this whole mess. 

I just want to make sure we have good test coverage on KoreaderHelper.

Reply:
Understandably. I think the underlying issue is that we are trying to mock out static functionality (reading a file) using a concrete object.

I haven't thought this through all the way, but what if we added the mocking capability to a static object? It obviously has its drawbacks...
```
public static class KavitaFile
{
    private Dictionary<string, Stream> mockedFiles;

    public static Stream OpenRead(string path)
    {
        if (mockedFiles.Any())
        {
            return File.OpenRead(path);
        }
        else
        {
            return mockedFiles[path];
        }
    }

    public static void AddMockedFile(string path, Stream stream)
    {
        mockedFiles.Add(path, stream);
    }
}
```
This is incomplete but gets the idea across. Before any tests run, you need to set up the predefined mock files. It makes sense to me that you'd only want one mocking strategy, so I'm assuming you'd need to convert all places that you're using the directory service.
The most obvious drawback is abusing the mock functionality in application code. I don't think I'd include this in the current PR. Is this a route worth pursuing?

Otherwise, I can grab a few books of Project Gutenberg for test data.

Reply:
I think let's just grab some files. Moving to Manga files adds a lot of mess and really not needed. As long as we can test, I'm happy.